### PR TITLE
[MOSTLY MODULAR] Mediguns: The start of utility cells. 

### PR DESCRIPTION
**Pull request by softcerv**
## About The Pull Request
Adds in two new utility class Medicells to the game, this will be the start of giving the medigun more functions that aren't raw healing. Both of these function similarly to preexisting chems, but are significantly weaker and more suited towards use that wouldn't otherwise merit using chems or in cases where they can't be chems can't easily be obtained in the first place. Both of these are unlocked by Advanced Bio research, similar to Tier II Medicells

## Clotting Medicell
coagulant
This cell functions as a coagulant. Infusing the patient with a weaker version of Sanguirite (1/2 the clotting speed), along with Iron. Great for repairing bleeding that wouldn't merit a medipen. 
## Temperature Adjustment Medicell
temp
The temperature adjustment medicell functions by increasing (or decreasing) a patients temperature towards the normal for their body by around 30k with each shot. This would be good in scenarios like warming up people post cryo, but would struggle in scenerios with sustained temperature damage, being less efficient and requiring more attention then using Leporazine. 

## How This Contributes To The Skyrat Roleplay Experience
These should be a good step towards giving Mediguns their own unique niche, and should set a good framework for future utility medicells.  

## Changelog
:cl:
expansion: Adds in two new Medicells, please read "https://github.com/Skyrat-SS13/Skyrat-tg/pull/8008" for more information
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/modular_reagents/code/reagents/medicine.dm [hunk 1]
```diff
@@ -43,3 +43,12 @@
 	owner.adjustStaminaLoss(5 * REM * delta_time, 0)
 	if(prob(30))
 		INVOKE_ASYNC(owner, /mob.proc/emote, "scream")
+
+//Medigun Clotting Medicine
+/datum/reagent/medicine/coagulant/fabricated
+	name = "Fabricated Coagulant"
+	description = "A synthesized cogaulant created by Mediguns."
```
Comment (tf-4) [inline modular_skyrat/modules/modular_reagents/code/reagents/medicine.dm]:
```suggestion
	description = "A synthesized coagulant created by Mediguns."
```

File: modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm [hunk 2]
```diff
@@ -297,5 +299,47 @@
...
+
+/obj/projectile/energy/medical/utility/clotting
+	name = "clotting agent shot"
+
+/obj/projectile/energy/medical/utility/clotting/on_hit(mob/living/target)
+	if(!IsLivingHuman(target))
+		return FALSE
+	if(target.reagents.get_reagent_amount(/datum/reagent/medicine/coagulant/fabricated) < 5) //injects the target with a weaker coagulant agent
+		target.reagents.add_reagent(/datum/reagent/medicine/coagulant/fabricated, 1)
+		target.reagents.add_reagent(/datum/reagent/iron, 2) //adds in iron to help conpensate for the relatively weaker blood clotting
```
Comment (tf-4) [inline modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm]:
```suggestion
		target.reagents.add_reagent(/datum/reagent/iron, 2) //adds in iron to help compensate for the relatively weak blood clotting
```

File: modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm [hunk 3]
```diff
@@ -297,5 +299,47 @@
...
+	select_name = "temperature"
+
+/obj/projectile/energy/medical/utility/temperature
+	name = "temperature adjustment shot"
 
+/obj/projectile/energy/medical/utility/temperature/on_hit(mob/living/target)
+	if(!IsLivingHuman(target))
+		return FALSE
+	var/ideal_temp = target.get_body_temp_normal(apply_change=FALSE) //Gets the temperature we should be aiming for.
+	var/current_temp = target.bodytemperature //Retrives the targets body temperature
```
Comment (tf-4) [inline modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm]:
```suggestion
	var/current_temp = target.bodytemperature //Retrieves the targets body temperature
```

File: modular_skyrat/modules/modular_reagents/code/reagents/medicine.dm [hunk 4]
```diff
@@ -43,3 +43,12 @@
 	owner.adjustStaminaLoss(5 * REM * delta_time, 0)
 	if(prob(30))
 		INVOKE_ASYNC(owner, /mob.proc/emote, "scream")
+
+//Medigun Clotting Medicine
+/datum/reagent/medicine/coagulant/fabricated
+	name = "Fabricated Coagulant"
```
Comment (tf-4) [inline modular_skyrat/modules/modular_reagents/code/reagents/medicine.dm]:
```suggestion
	name = "fabricated coagulant"
```
No need to capitalise the names

File: modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medguns.dm [hunk 5]
```diff
@@ -187,6 +188,24 @@
 	icon_state = "Oxy3"
 	ammo_type = /obj/item/ammo_casing/energy/medical/oxy3
 //End of Tier III
+//Start of Utility Cells
+/obj/item/medicell/utility
+	utility = TRUE
+	name = "Utility Class Medicell"
```
Comment (tf-4) [inline modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medguns.dm]:
```suggestion
	name = "utility class medicell"
```

File: modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medguns.dm [hunk 6]
```diff
@@ -187,6 +188,24 @@
...
+	desc = "You really shouldn't be seeing this, if you do, please yell on GitHub."
+
+/obj/item/medicell/utility/clotting
+	name = "Clotting Medicell"
```
Comment (tf-4) [inline modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medguns.dm]:
```suggestion
	name = "clotting medicell"
```

File: modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medguns.dm [hunk 7]
```diff
@@ -187,6 +188,24 @@
...
+	desc = "A medicell designed to help deal with bleeding patients"
+	icon_state = "clotting"
+	ammo_type = /obj/item/ammo_casing/energy/medical/utility/clotting
+
+/obj/item/medicell/utility/temperature
+	name = "Temperature Readjustment Medicell"
```
Comment (tf-4) [inline modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medguns.dm]:
```suggestion
	name = "temperature readjustment medicell"
```

File: modular_skyrat/modules/modular_weapons/code/modules/research/designs/mediguns.dm [hunk 8]
```diff
@@ -87,6 +87,30 @@
 	reagents_list = list(/datum/reagent/medicine/salbutamol = 10)
 	category = list("Ammo")
 	departmental_flags = DEPARTMENTAL_FLAG_MEDICAL
+
+//Utility Medicells
+/datum/design/clotmedicell
+	name = "Clotting Medicell"
```
Comment (tf-4) [inline modular_skyrat/modules/modular_weapons/code/modules/research/designs/mediguns.dm]:
```suggestion
	name = "clotting medicell"
```

File: modular_skyrat/modules/modular_weapons/code/modules/research/designs/mediguns.dm [hunk 9]
```diff
@@ -87,6 +87,30 @@
...
+	id = "clotmedicell"
+	build_type = PROTOLATHE | AWAY_LATHE
+	materials = list(/datum/material/plastic = 2000, /datum/material/glass = 2000, /datum/material/plasma = 1000, /datum/material/diamond = 500)
+	build_path = /obj/item/medicell/utility/clotting
+	reagents_list = list(/datum/reagent/medicine/salglu_solution = 5, /datum/reagent/blood = 5)
+	category = list("Ammo")
+	departmental_flags = DEPARTMENTAL_FLAG_MEDICAL
+
+/datum/design/tempmedicell
+	name = "Temperature Adjustment Medicell"
```
Comment (tf-4) [inline modular_skyrat/modules/modular_weapons/code/modules/research/designs/mediguns.dm]:
```suggestion
	name = "temperature adjustment medicell"
```

File: modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm [hunk 10]
```diff
@@ -297,5 +299,47 @@
...
 
+/obj/projectile/energy/medical/utility/temperature/on_hit(mob/living/target)
+	if(!IsLivingHuman(target))
+		return FALSE
+	var/ideal_temp = target.get_body_temp_normal(apply_change=FALSE) //Gets the temperature we should be aiming for.
+	var/current_temp = target.bodytemperature //Retrieves the targets body temperature
+	var/difference = ideal_temp - current_temp
+	if(abs(difference) <= MINIMUM_TEMP_DIFFERENCE) //It won't adjust temperature if the difference is too low
+		return FALSE
+	target.adjust_bodytemperature(difference < 0 ? -TEMP_PER_SHOT : TEMP_PER_SHOT)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm:341]: Just to make it look nicer.
```suggestion
	target.adjust_bodytemperature(difference < 0 ? -TEMP_PER_SHOT : TEMP_PER_SHOT)

```

File: modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medguns.dm [hunk 11]
```diff
@@ -187,6 +188,24 @@
 	icon_state = "Oxy3"
 	ammo_type = /obj/item/ammo_casing/energy/medical/oxy3
 //End of Tier III
+//Start of Utility Cells
+/obj/item/medicell/utility
+	utility = TRUE
+	name = "utility class medicell"
+	desc = "You really shouldn't be seeing this, if you do, please yell on GitHub."
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medguns.dm]:
```suggestion
	desc = "You really shouldn't be seeing this, if you do, please yell at your local coders."
```

**Review by GoldenAlpharex — APPROVED** [review]
Aside from those two tiny comments, I'd say this is pretty solid, I'll let someone else take a look if they want, but it seems functional to me.

**Review by GoldenAlpharex — APPROVED** [review]

**Pull request merged**